performance_regression_test.py

```diff
@@ -797,3 +798,100 @@ def build_histogram(self, workload: PerformanceTestWorkload, test_type: Performa
 
         self.update_hdrhistograms(histogram_name='test_histogram_by_interval',
                                   histogram_data=histogram_data_by_interval)
+
+
+class PerformanceRegressionUpgradeTest(PerformanceRegressionTest, UpgradeTest):  # pylint: disable=too-many-ancestors
+    def __int__(self, *args):
+        super().__init__(*args)
+        self.cluster = None  # pylint: disable=attribute-defined-outside-init
+        self.tester = None  # pylint: disable=attribute-defined-outside-init
+        self.monitoring_set = None  # pylint: disable=attribute-defined-outside-init
+
+    def get_email_data(self):  # pylint: disable=no-self-use
+        return PerformanceRegressionTest.get_email_data(self)
+
+    @latency_calculator_decorator(legend="Upgrade Node")
+    def upgrade_node(self, node):  # pylint: disable=arguments-differ
+        InfoEvent(message='Upgrade Node %s begin' % node.name).publish()
+        self._upgrade_node(node)
+        InfoEvent(message='Upgrade Node %s ended' % node.name).publish()
+
+    def _stop_stress_when_finished(self):  # pylint: disable=no-self-use
+        with EventsSeverityChangerFilter(new_severity=Severity.NORMAL,  # killing stress creates Critical error
+                                         event_class=CassandraStressEvent,
+                                         extra_time_to_expiration=60):
+            self.loaders.kill_stress_thread()
+
+    @latency_calculator_decorator
+    def steady_state_latency(self):  # pylint: disable=no-self-use
+        sleep_time = self.db_cluster.params.get('nemesis_interval') * 60
+        InfoEvent(message='Starting Steady State calculation for %ss' % sleep_time).publish()
+        time.sleep(sleep_time)
+        InfoEvent(message='Ended Steady State calculation. Took %ss' % sleep_time).publish()
+
+    @latency_calculator_decorator
+    def post_upgrades_steady_state(self):
```

Comment:
log_run_info

Reply:
won't do it, as with using `log_run_info` we won't see it annotated in Grafana